src/debug/configurationProvider.ts

```diff
@@ -118,9 +118,15 @@ export class ArduinoDebugConfigurationProvider implements vscode.DebugConfigurat
         const dc = DeviceContext.getInstance();
 
         if (!config.program || config.program === "${file}") {
-            // make a unique temp folder because keeping same temp folder will corrupt the build when board is changed
-            const outputFolder = path.join(dc.output || `.build`, ArduinoContext.boardManager.currentBoard.board);
-            util.mkdirRecursivelySync(path.join(ArduinoWorkspace.rootPath, outputFolder));
+            const outputFolder = path.join(dc.output || `.build`);
+            const outputPath = path.join(ArduinoWorkspace.rootPath, outputFolder);
+            
+             // if the directory was already there, clear the folder so that it's not corrupted from previous builds.

```

Comment:
I'm guessing this won't affect too many users, but just to clarify: this means that a user that frequently switches between boards will also be forced to frequently rebuild, correct?

Reply:
Yes, that would be the case.